wear/src/main/kotlin/au/com/shiftyjelly/pocketcasts/wear/ui/LoggingInScreen.kt

```diff
@@ -80,17 +85,57 @@ fun LoggingInScreen(
             style = MaterialTheme.typography.title3
         )
         if (email != null) {
-            Text(
-                text = email,
-                textAlign = TextAlign.Center,
-                maxLines = 1,
-                overflow = TextOverflow.Ellipsis,
-                style = MaterialTheme.typography.body2
-            )
+            BoxWithConstraints(
+                contentAlignment = Alignment.Center,
+                modifier = Modifier
+                    .fillMaxWidth()
+            ) {
+
+                val widthPx = with(LocalDensity.current) { maxWidth.toPx() }
+                val background = MaterialTheme.colors.background
+
+                Text(
+                    text = email,
+                    style = MaterialTheme.typography.body2,
+                    // Turn off softWrap to make sure the text doesn't get truncated if it runs long.
+                    // Without this if "xxxx@gmail.com" ran just a bit long, it would get shortened
+                    // to "xxx@gmail".
+                    softWrap = false,
+                    modifier = Modifier.fadeOutOverflow(
+                        overFlowWidthPx = widthPx.toInt(),
+                        backgroundColor = background
+                    )
+                )
+            }
         }
     }
 }
 
+private fun Modifier.fadeOutOverflow(
+    overFlowWidthPx: Int,
+    backgroundColor: Color,
+) = drawWithContent {
+
+    // draw text first
+    drawContent()
+
+    // Use < instead of <= because text that doesn't fit will have a width that is equal
+    // to the available space
+    val textFits = size.width < overFlowWidthPx
```

Comment:
This does mean that if the text _exactly_ fits on the screen, the overflow gradient will get applied. It seems that no matter how long the text is it will not report a `size.width` that is larger than the `BoxWithConstraint`'s width, so basically if they are equal then I'm assuming overflow. That's correct most of the time, but it does create an edge case for text that is extremely close to overflowing.